lib/ash/actions/create/bulk.ex

```diff
@@ -1216,12 +1275,38 @@ defmodule Ash.Actions.Create.Bulk do
             end
 
           case result do
+            {:ok, {:affected, affected}} ->
+              Process.put({:any_success?, ref}, true)
+
+              store_batch_status(
+                ref,
+                %Ash.BulkResult.BatchStatus{
+                  inputs: length(batch),
+                  affected: affected
+                },
+                opts
+              )
+
+              []
+
             {:ok, result} ->
               Process.put({:any_success?, ref}, true)
               result
 
             :ok ->
               Process.put({:any_success?, ref}, true)
+
+              cond do
+                opts[:return_records?] ->
+                  raise "`#{inspect(Ash.Resource.Info.data_layer(resource))}.bulk_create/3` returned :ok without a result when `return_records?` is true"
+
+                opts[:return_batch_status?] ->
+                  raise "`#{inspect(Ash.Resource.Info.data_layer(resource))}.bulk_create/3` returned :ok without affected row count when `return_batch_status?` is true"
```

Comment:
We should synthesize this return type for calls to single creates.

Reply:
As I understand the current code, it is fully synthesized. The error is only raised if you implement `bulk_create` and return `:ok` from there.

Am I missing something?

That fn could use a refactor, it's quite hard to follow what it does :sweat_smile:

Reply:
Ah, right because that callback returns `{:ok, result}`.